src/api/composition/CompositionProvider.js

```diff
@@ -19,32 +19,48 @@
  * this source code distribution or the Licensing information page available
  * at runtime from the About dialog for additional information.
  *****************************************************************************/
+import _ from 'lodash';
+import objectUtils from "../objects/object-utils";
 
-define([
-    'lodash',
-    'objectUtils'
-], function (
-    _,
-    objectUtils
-) {
+/**
+ * @typedef {import('../objects/ObjectAPI').DomainObject} DomainObject
+ */
+
+/**
+ * @typedef {import('../objects/ObjectAPI').Identifier} Identifier
+ */
+
+/**
+ * @typedef {import('./CompositionAPI').default} CompositionAPI
+ */
+
+/**
+ * @typedef {import('../../../openmct').OpenMCT} OpenMCT
+ */
+
+/**
+ * A CompositionProvider provides the underlying implementation of
+ * composition-related behavior for certain types of domain object.
+ *
+ * By default, a composition provider will not support composition
+ * modification.  You can add support for mutation of composition by
+ * defining `add` and/or `remove` methods.
+ *
+ * If the composition of an object can change over time-- perhaps via
+ * server updates or mutation via the add/remove methods, then one must
+ * trigger events as necessary.
+ *
+ * @memberof module:openmct
+ */
+export default class CompositionProvider {
```

Comment:
In order for type `CompositionProvider` to be generated, the class must be named the same. So change `DefaultCompositionProvider` --> `CompositionProvider` and convert to ES6 class

Reply:
They're not really the same thing. `CompositionProvider` is intended to represent an interface, `DefaultCompositionProvider` is an implementation of that interface which provides composition via the `.composition` attribute - which is just one way of defining compositional relationships.

I think the right way to do this is to define a `CompositionProvider` class where the docs are declared, and then have DefaultCompositionProvider extend it. I don't know if there's some "standard" way of declaring a class "abstract" in ES6, but in this situation in the past I have just returned an Error from all of the functions on the class I want to be "abstract".

Reply:
OK, good catch. I think the method you mentioned for "hacking" ES6 abstract classes *should* work, I'll give it a go

Reply:
@akhenry Please take a look at how I've implemented this. It's not ideal, as we access private members within (what I think should be) non-private methods. As such I've declared accessors for the specific private members which would be used in the derived class. Which just seems like I'm exposing private members in a roundabout way? It feels weird...

The other not-so-nice thing is that JSDoc/TypeScript doesn't inherit the docs from the base class! Totally lame! So we have to declare them in two places! There's apparently a fix for this in the works, but /shrug.

Let me know what you think and how we should move forward. Docs or no docs, I think we should do what is best structurally for the API.